firestore/src/main/transaction_main.cc

```diff
@@ -156,5 +163,14 @@ void TransactionInternal::MarkPermanentlyFailed() {
   transaction_->MarkPermanentlyFailed();
 }
 
+void TransactionInternal::ValidateReference(const DocumentReference& document) {
+  auto* internal_doc = GetInternal(&document);
```

Comment:
Isn't it possible that the "internal" DocumentReference is nullptr? If so, you should probably check if it's null before using it.

Reply:
Good catch. The existing code would crash in that case. I'll add a validation test to cover this tomorrow.

Reply:
Just waiting for this to be addressed. Then I'll approve.

Reply:
Added the test (also addressed the same problem with `WriteBatch`).